src/backends/github.rs

```diff
@@ -340,9 +346,10 @@ impl UpdateBuilder {
 
     /// Set the exe's name. Also sets `bin_path_in_archive` if it hasn't already been set.
     pub fn bin_name(&mut self, name: &str) -> &mut Self {
-        self.bin_name = Some(name.to_owned());
+        let raw_bin_name = format!("{}{}", name, EXECUTABLE_SUFFIX);
```

Comment:
could this check the `name` and only add the suffix when it's missing?

Reply:
I think it would make more sense to keep it like that. The `format!()` call should result in the same as code as `to_owned()` when the suffix is empty; the compiler is probably able to optimize this away since `EXECUTABLE_SUFFIX`/`std::env::consts::EXE_SUFFIX` are constants.

So I think the proposed code is clearer _and_ produces optimal machine code. Adding extra checks would complicate things without much benefit.

Reply:
That makes sense! I'm not as worried about allocations as I am that the this method will be called with the wrong string -- if you call `bin_name("my.exe")` on windows, will that result in `raw_bin_name == "my.exe.exe"` ?

Reply:
Good point!

Looking at #21 I realized that the `.exe` _can_ appear in the `bin_name()` call... So I simply used `bin_name("my.exe")` and it worked.

The `name` could always be filtered of the `.exe` before being concatenated with it again. Something like:
```
let raw_bin_name = format!("{}{}", name.trim_end(std::env::consts::EXE_SUFFIX), EXECUTABLE_SUFFIX);
```

I'm having trouble running my own Windows CI-built releases but I'll try it after...